### PR TITLE
Feature/agent 353 spark 3 2 support

### LINKED ISSUE #353: Spark 3.2 support
Check the changes in the Spark internal API and the dependecies, and create a separate bundle for Spark 3.2 if 3.1 doesn't suit

### PR DESCRIPTION
**Pull request by cerveada**
- compare URIs as URIs not as strings
- replace init with stripSuffix("/") to clarify the intention
- exclude netty dependencies of zookeeper
- use different json4s versions for spark 3.1 and 3.2
- rework LineageHarvesterSpec

### CODE REVIEW

File: integration-tests/src/main/scala/za/co/absa/spline/test/SplineMatchers.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      MatchResult(
+        URI.create(left) == URI.create(expectedUri),
+        s"URI $left was not the same as expected $expectedUri",
+        s"URI $left was the same as expected $expectedUri"
+      )
+    }
+  }
+
+  def beSameUriAs(expectedUri: String) = new SameUriAs(expectedUri)
```
Comment (wajda) [inline integration-tests/src/main/scala/za/co/absa/spline/test/SplineMatchers.scala]: I think it's worth to be included in the Commons. I would only rename the method to `equalToUri`, as it communicates semantics more precisely.

File: pom.xml [hunk 2]
```diff
@@ -125,8 +128,8 @@
         <scala_2.12.version>2.12.15</scala_2.12.version>
 
         <!-- Controlled by `scala-cross-build` plugin -->
-        <scala.version>2.11.12</scala.version>
-        <scala.binary.version>2.11</scala.binary.version>
+        <scala.version>2.12.15</scala.version>
+        <scala.binary.version>2.12</scala.binary.version>
```
Comment (wajda) [inline pom.xml]: Are we switching to Scala 2.12 as the default?

**Comment by wajda** [comment]
I have no idea why CI tests are passing, but I've got this
```
*** RUN ABORTED ***
  java.lang.NoSuchMethodError: io.netty.buffer.PooledByteBufAllocator.defaultNumHeapArena()I
  at org.apache.spark.network.util.NettyUtils.createPooledByteBufAllocator(NettyUtils.java:113)
  at org.apache.spark.network.client.TransportClientFactory.<init>(TransportClientFactory.java:106)
  at org.apache.spark.network.TransportContext.createClientFactory(TransportContext.java:99)
  at org.apache.spark.rpc.netty.NettyRpcEnv.<init>(NettyRpcEnv.scala:71)
  at org.apache.spark.rpc.netty.NettyRpcEnvFactory.create(NettyRpcEnv.scala:461)
  at org.apache.spark.rpc.RpcEnv$.create(RpcEnv.scala:57)
  at org.apache.spark.SparkEnv$.create(SparkEnv.scala:249)
  at org.apache.spark.SparkEnv$.createDriverEnv(SparkEnv.scala:175)
  at org.apache.spark.SparkContext.createSparkEnv(SparkContext.scala:257)
  at org.apache.spark.SparkContext.<init>(SparkContext.scala:424)
  ...
```
Something must be wrong with the Netty dependency.

File: pom.xml (hunk 2 above)
Comment (cerveada) [reply pom.xml]: I somehow committed this by mistake, I will revert it.

**Review by wajda — DISMISSED** [review]
LGTM

**Review by wajda — APPROVED** [review]

**Pull request merged**